esx_service/vmci/vmci_client.c

```diff
@@ -291,6 +291,7 @@ vsock_get_reply(be_sock_id *s, be_request *r, be_answer* a)
    ret = recv(s->sock_id, a->buf, b, 0);
    if (ret == -1 || ret != b) {
       free(a->buf);
+      a->buf = NULL;
```

Comment:
nit - well, just to be paranoid, it should be 
```C
   if (a->buf) {
     free(a->buf);
     a->buf = NULL;
  }
```

I know a->buf is not null (per L286-L289, but just to keep this section easier to read and more resilient to changes above it :-)